src/domUpdates.js

```diff
@@ -68,13 +92,55 @@ let domUpdates = {
     userGreeting.html(`Welcome, ${data.name}!!!`)
   },
 
+  greetAgent() {
+    userGreeting.html(`Welcome, Agent!!!`);
+    tripsHeader.text(`Trips pending Approval:`);
+    tripsHeader.toggleClass('hidden');
+    spendHeader.toggleClass('hidden');
+  },
+
   displayTrips(trips) {
     let tripsToDisplay = trips.map((trip) => {
       return `<li>${trip.destination.destination} - ${trip.status}</li>`
     }).join('')
     tripsHeader.toggleClass('hidden');
-    tripsStatus.html(tripsToDisplay)
+    tripsStatus.html(tripsToDisplay);
+  },
+
+  displayAllPendingTrips(allTravelers) {
+    // get all pending trips
+    // const allTravelersWithPendingTrips = this.getAllTravelersWithPendingTrips(allTravelers)
+    //generate html elements
+    const allPendingTripHtml = allTravelers.reduce((acc1, traveler) => {
+      const allTripDataForTraveler = traveler.trips.reduce((acc2, trip) => {
+        if (trip.status === 'pending') {
+          const listElement = `<li>${traveler.name} - ${trip.destination.destination} - ${trip.status} <button class='approve-trip' type='button'>Approve</button><button class='deny-trip' type='button'>Deny</button></li>`
+          acc2 += listElement
```

Comment:
I totally name my accumulators acc as placeholders all the time, but I think the instructors want us to semantically name them. Just a quick fix you could do before submitting your project.